script/NestedSignFromJson.s.sol

```diff
@@ -8,25 +8,30 @@ import {stdJson} from "forge-std/StdJson.sol";
 import {console} from "forge-std/console.sol";
 
 contract NestedSignFromJson is NestedMultisigBuilder, JsonTxBuilderBase {
+    address globalSignerSafe; // Hack to avoid passing signerSafe as an input to many functions.
+
     /// @dev Signs the approveHash transaction from the Nested Safe to the System Owner Safe.
     function signJson(string memory _path, address _signerSafe) public {
         _loadJson(_path);
         sign(_signerSafe);
-        _postCheck();
+        globalSignerSafe = _signerSafe;
+        _postCheckWithSim();
     }
 
     /// @dev Submits signatures to call approveHash on the System Owner Safe.
     function approveJson(string memory _path, address _signerSafe, bytes memory _signatures) public {
         _loadJson(_path);
         approve(_signerSafe, _signatures);
-        _postCheck();
+        globalSignerSafe = _signerSafe;
+        _postCheckWithSim();
     }
 
     /// @dev Executes the transaction from the System Owner Safe.
     function runJson(string memory _path) public {
         _loadJson(_path);
         run();
-        _postCheck();
+        // globalSignerSafe = _signerSafe; // TODO how to handle this one?
```

Comment:
Could `vm.env` work?

Reply:
Good idea. Since this code path isn't run until actual execution of the signed transaction, let's merge this and I'll look into this